src/main/kotlin/it/pagopa/wallet/services/WalletService.kt

```diff
@@ -463,6 +467,48 @@ class WalletService(
             }
     }
 
+    private fun handleWalletNotification(
+        wallet: Wallet,
+        walletNotificationRequestDto: WalletNotificationRequestDto
+    ): Pair<WalletStatusDto, it.pagopa.wallet.domain.details.WalletDetails<*>> {
+        val operationResult = walletNotificationRequestDto.operationResult
+        val operationDetails = walletNotificationRequestDto.details
+        logger.info(
+            "Received wallet notification request for wallet with id: [{}]. Outcome: [{}], notification details: [{}]",
+            wallet.id.value,
+            operationResult,
+            operationDetails
+        )
+        return when (val walletDetails = wallet.details) {
+            is it.pagopa.wallet.domain.details.CardDetails ->
+                if (operationResult == WalletNotificationRequestDto.OperationResultEnum.EXECUTED) {
+                    Pair(WalletStatusDto.VALIDATED, walletDetails)
+                } else {
+                    Pair(WalletStatusDto.ERROR, walletDetails)
+                }
+            is it.pagopa.wallet.domain.details.PayPalDetails ->
+                if (operationResult == WalletNotificationRequestDto.OperationResultEnum.EXECUTED) {
+                    if (operationDetails is WalletNotificationRequestPaypalDetailsDto) {
+                        Pair(
+                            WalletStatusDto.VALIDATED,
+                            walletDetails.copy(maskedEmail = operationDetails.maskedEmail)
+                        )
+                    } else {
+                        logger.error(
+                            "No details received for PayPal wallet, cannot retrieve maskedEmail"
+                        )
+                        Pair(WalletStatusDto.ERROR, walletDetails)
```

Comment:
Isn't it better to throw an exception in this case? throw InvalidRequestException() ?

Reply:
But this way wallet will remain in `VALIDATION_REQUESTED` state. Instead doing this wallet will go to ERROR for distinguish cases where wallet did not receive POST notify from those where we receive a POST notify with erroneous data.

Reply:
Ok it comes back to me even if hypothetically it's as if we are receiving a request with the wrong body right? In that case we would have responded with a 400 without changing the status of the wallet

Reply:
![immagine](https://github.com/pagopa/pagopa-wallet-service/assets/8624484/d52500be-65f2-4c33-9559-164da614e069)

Reply:
Fixed with https://github.com/pagopa/pagopa-wallet-service/pull/76/commits/ac62c1a3ef0617f644a3393606ab9d1dac637e10